system/lib/libcxxabi/src/format_exception.cpp

```diff
@@ -1,39 +1,46 @@
 #include "cxa_exception.h"
-#include "cxxabi.h"
+#include "private_typeinfo.h"
 #include <stdio.h>
-#include <typeinfo>
 
 #ifdef __USING_EMSCRIPTEN_EXCEPTIONS__
 
-extern "C" {
+using namespace __cxxabiv1;
+
+//  Utility routines copied from cxa_exception.cpp
+static inline __cxa_exception*
+cxa_exception_from_thrown_object(void* thrown_object) {
+  return static_cast<__cxa_exception*>(thrown_object) - 1;
+}
```

Comment:
I have another change that also adds this function: https://github.com/emscripten-core/emscripten/pull/16627

But I don't mind merging them when I land that.